### PR TITLE
Feature: Improved shell extension loading

### LINKED ISSUE #11882: Feature: Improve shell extension loading performance
### What feature or improvement do you think would benefit Files?

`EnumMenuItems()` in `ContextMenu` class is synchronous and recursive, which is why it takes so long to display shell extensions.
Changing this should help improve performance.

### Requirements

Avoid synchronous and recursive calling in `GetContextMenuForFiles()` and `EnumMenuItems()` by:
* Separate the loading process for the main menu and submenus
* Call `EnumMenuItems()` asynchronously if possible

### Files Version

2.4.58.0

### Windows Version

Windows 11 22H2

### Comments

_No response_

### PR DESCRIPTION
**Pull request by hishitetsu**
It loads main menu first, then loads Open with submenu, Send to submenu and other submenus in order.

**Resolved / Related Issues**
- [X] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #11882 

**Validation**
How did you test these changes?
- [X] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

**GIF (2.4.60 is light, this PR is dark)**
無題の動画 ‐ Clipchampで作成

### CODE REVIEW

**Comment by yaira2** [comment]
Right clicking a folder with media files is causing the window to become unresponsive

File: src/Files.App/BaseLayout.cs [hunk 1]
```diff
@@ -717,12 +718,13 @@ private void AddNewFileTagsToMenu(CommandBarFlyout contextMenu)
 			});
 		}
 
-		private void AddShellItemsToMenu(List<ContextMenuFlyoutItemViewModel> shellMenuItems, CommandBarFlyout contextMenuFlyout, bool shiftPressed)
+		private async Task AddShellItemsToMenu(List<ContextMenuFlyoutItemViewModel> shellMenuItems, CommandBarFlyout contextMenuFlyout, bool shiftPressed)
```
Comment (lukeblevins) [inline src/Files.App/BaseLayout.cs:721]:
```suggestion
		private async Task AddShellMenuItemsAsync(List<ContextMenuFlyoutItemViewModel> shellMenuItems, CommandBarFlyout contextMenuFlyout, bool shiftPressed)
```

File: src/Files.App/BaseLayout.cs [hunk 2]
```diff
@@ -670,7 +671,7 @@ private async Task LoadMenuItemsAsync()
 			{
 				var shellMenuItems = await ContextFlyoutItemHelper.GetItemContextShellCommandsAsync(workingDir: ParentShellPageInstance.FilesystemViewModel.WorkingDirectory, selectedItems: SelectedItems!, shiftPressed: shiftPressed, showOpenMenu: false, shellContextMenuItemCancellationToken.Token);
 				if (shellMenuItems.Any())
-					AddShellItemsToMenu(shellMenuItems, ItemContextMenuFlyout, shiftPressed);
+					await AddShellItemsToMenu(shellMenuItems, ItemContextMenuFlyout, shiftPressed);
```
Comment (lukeblevins) [inline src/Files.App/BaseLayout.cs:674]:
```suggestion
					await AddShellMenuItemsAsync(shellMenuItems, ItemContextMenuFlyout, shiftPressed);
```

File: src/Files.App/BaseLayout.cs [hunk 3]
```diff
@@ -611,7 +612,7 @@ public async void BaseContextFlyout_Opening(object? sender, object e)
 				{
 					var shellMenuItems = await ContextFlyoutItemHelper.GetItemContextShellCommandsAsync(workingDir: ParentShellPageInstance.FilesystemViewModel.WorkingDirectory, selectedItems: new List<ListedItem>(), shiftPressed: shiftPressed, showOpenMenu: false, shellContextMenuItemCancellationToken.Token);
 					if (shellMenuItems.Any())
-						AddShellItemsToMenu(shellMenuItems, BaseContextMenuFlyout, shiftPressed);
+						await AddShellItemsToMenu(shellMenuItems, BaseContextMenuFlyout, shiftPressed);
```
Comment (lukeblevins) [inline src/Files.App/BaseLayout.cs:615]:
```suggestion
						await AddShellMenuItemsAsync(shellMenuItems, BaseContextMenuFlyout, shiftPressed);
```

**Review by lukeblevins — CHANGES_REQUESTED** [review]
Looks good! We just need to rename that method since it's async now

**Comment by hishitetsu** [comment]
> Right clicking a folder with media files is causing the window to become unresponsive

In my environment, I can right-click a folder with some screenshots normally.
Doesn't it become unresponsive in 2.4.60?

**Comment by yaira2** [comment]
> In my environment, I can right-click a folder with some screenshots normally.

Can you try with music files?

> Doesn't it become unresponsive in 2.4.60?

It works fine in v2.4.60 but perhaps that's because it's in release mode.

**Comment by hishitetsu** [comment]
> Can you try with music files?

It works fine for me.

**Review by yaira2 — APPROVED** [review]
LGTM

**Review by lukeblevins — APPROVED** [review]

**Pull request merged**